### DISCUSSION TITLE
[@types/xrm] How to use the XrmEnum?

### DISCUSSION THREAD

**Discussion by ac1223** [DefinitelyTyped/DefinitelyTyped#57204] [Issues with a @types package]
Hi,

I'm trying to use the XrmEnum.FormNotificationLevel.Error enum to set the FormNotificationLevel param of the formContext.ui.setFormNotification function. All compiles well, but when deployed to Dynamics 365, it is throwing an error saying "Uncaught ReferenceError: XrmEnum is not defined".

How am i supposed to be able to use the XrmEnum's?

[a comment by an automated account is not shown]

**Comment by ac1223** [comment]
Updated. Thanks.

**Reply by daryllabar** [reply]
There is no enum:
```
    /**
     * Form Notification Levels for {@link Ui.setFormNotification formContext.ui.setFormNotification()}.
     * @see {@link XrmEnum.FormNotificationLevel}
     */
    type FormNotificationLevel = "ERROR" | "INFO" | "WARNING";

    interface Ui {
        /**
         * Displays a form level notification. Any number of notifications can be displayed and will remain until removed using clearFormNotification.
         * The height of the notification area is limited so each new message will be added to the top.
         * @param message The text of the notification message.
         * @param level The level of the notification which defines how the message will be displayed, such as the icon.
         * ERROR: Notification will use the system error icon.
         * WARNING: Notification will use the system warning icon.
         * INFO: Notification will use the system info icon.
         * @param uniqueId Unique identifier for the notification which is used with clearFormNotification to remove the notification.
         * @returns true if it succeeds, otherwise false.
         */
        setFormNotification(message: string, level: FormNotificationLevel, uniqueId: string): boolean;
```
It's a string type, so it you have to pass in either a string of "ERROR" or "INFO" or "NEVER"

**Reply by ac1223** [reply]
Thankyou for your reply. I was referring to line 5481 of https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/xrm/index.d.ts

     /**
     * Constant Enum: Form Notification Levels for {@link Ui.setFormNotification formContext.ui.setFormNotification()}.
     * @see {@link Xrm.FormNotificationLevel}
     */
    const enum FormNotificationLevel {
        Error = "ERROR",
        Info = "INFO",
        Warning = "WARNING"
    }

Am I using it incorrectly here?
formContext.ui.setFormNotification("Alert text", **XrmEnum.FormNotificationLevel.Warning**, "abc");
